sympy/physics/vector/tests/test_frame.py

```diff
@@ -471,3 +471,29 @@ def test_orient_quaternion():
     B = ReferenceFrame('B')
     B.orient_quaternion(A, (0,0,0,0))
     assert B.dcm(A) == Matrix([[0, 0, 0], [0, 0, 0], [0, 0, 0]])
+
+def test_vector_cache():
+    A = ReferenceFrame('A')
+    B = ReferenceFrame('B')
+    C = ReferenceFrame('C')
+
+    a, b, c = symbols('a b c')
+
+    B.orient(A, 'Axis', (a, A.x))
+    assert A._dcm_dict[B] == Matrix([[1, 0, 0], [0, cos(a), -sin(a)], [0, sin(a), cos(a)]])
+    assert B._dcm_dict[A] == Matrix([[1, 0, 0], [0, cos(a), sin(a)], [0, -sin(a), cos(a)]])
+    assert C._dcm_dict == {}
+
+    B.orient(C, 'Axis', (b, C.x))
+    # Previous relation is not wiped
+    assert A._dcm_dict[B] == Matrix([[1, 0, 0], [0, cos(a), -sin(a)], [0, sin(a), cos(a)]])
+    assert B._dcm_dict[A] == Matrix([[1, 0, 0], [0, cos(a), sin(a)], [0, -sin(a), cos(a)]])
+    assert B._dcm_dict[C] == Matrix([[1, 0, 0], [0, cos(b), sin(b)], [0, -sin(b), cos(b)]])
+    assert C._dcm_dict[B] == Matrix([[1, 0, 0], [0, cos(b), -sin(b)], [0, sin(b), cos(b)]])
+
+    A.orient(B, 'Axis', (c, B.x))
+    # Previous relation is updated
+    assert B._dcm_dict[C] == Matrix([[1, 0, 0], [0, cos(b), sin(b)], [0, -sin(b), cos(b)]])
+    assert A._dcm_dict[B] == Matrix([[1, 0, 0], [0, cos(c), sin(c)], [0, -sin(c), cos(c)]])
+    assert B._dcm_dict[A] == Matrix([[1, 0, 0], [0, cos(c), -sin(c)], [0, sin(c), cos(c)]])
+    assert C._dcm_dict[B] == Matrix([[1, 0, 0], [0, cos(b), -sin(b)], [0, sin(b), cos(b)]])
```

Comment:
If you set B wrt A and then B wrt C, then overwrite A wrt B, B wrt C shouldn't change. It's not clear that is what is happening here.

I think these tests would be much clearer if you assert against the entire `_dcm_dict`. After each call to `orient`* we what the whole dictionary should look like, so test the whole dict not just specific entires.

Please also write some comments that explain what you are testing here. It's not clear without very careful look at all these sin's and cosines.

*Please use the new functions e.g. `orient_axis` in all future code.

Reply:
> If you set B wrt A and then B wrt C, then overwrite A wrt B, B wrt C shouldn't change. It's not clear that is what is happening here.

Yes this is happening here.

I'll update test cases to make it clearer.

Reply:
The unit tests should also have some tests that ensure the difference in the `_dcm_dict` and `_dcm_cache` is handled properly. Adding those tests with this change, would ensure we are breaking anything regarding that difference.

Reply:
Yes I will add seperate unit tests for `_dcm_cache` and `_dcm_dict`.

Reply:
One or more tests like:

```
B.orient(A)
C.orient(B)
D.orient(C)
# check that all _dcm_dicts have all adjacent relations properly defined and check that _dcm_cache == _dcm_dict
D.dcm(A)
# check that _dcm_cache's have the D <-> A relationship defined (but not _dcm_dict)
A.dcm(D)  # check to make sure result is pulled from the cache, not recomputed
A.orient(B)  # change orientation of A <-> B
# check that _dcm_cache has been wiped and now equals _dcm_dict
# etc
```